Decisions.Slack/Bots/Handlers/MentionHandler.cs

```diff
@@ -0,0 +1,81 @@
+using Decisions.Slack.Utility;
+using DecisionsFramework;
+using DecisionsFramework.Design.Flow;
+using DecisionsFramework.Design.Flow.Mapping;
+using DecisionsFramework.Design.Flow.Service.Execution;
+using DecisionsFramework.ServiceLayer.Services.ContextData;
+using SlackNet;
+using SlackNet.Events;
+using SlackNet.WebApi;
+
+namespace Decisions.Slack.Bots.Handlers;
+
+public class MentionHandler : IEventHandler<MessageEvent>
+{
+    private Log log = new Log(typeof(MentionHandler));
+    private readonly SlackBot _bot;
+    private readonly ISlackApiClient _api;
+    private readonly string _botUserId;
+    
+    public MentionHandler(SlackBot bot)
+    {
+        if (bot.ApiClient == null)
+        {
+            log.Error($"Failed to create ApiClient for SlackBot Id: {bot.GetEntityId()}");
+            return;
+        }
+
+        if (string.IsNullOrEmpty(bot.BotUserId))
+        {
+            log.Error($"Could not authenticate bot for SlackBot Id: {bot.GetEntityId()}");
+            return;
```

Comment:
Do you think it'd be better to throw here instead of just logging? If we're in this state, then it'll hit a NRE when it tries to do anything, right?